sdk/formrecognizer/ai-form-recognizer/src/documentAnalysisClient.ts

```diff
@@ -415,40 +427,215 @@ export class DocumentAnalysisClient {
       );
     }
 
-    return this.createAnalysisPoller<unknown>(input, {
-      initialModelId,
+    return this.createUnifiedPoller<unknown>(
+      (abortSignal) => {
+        const [contentType, analyzeRequest] = toAnalyzeRequest(input);
```

Comment:
Any chance we can make the return type of toAnalyzeRequest wrok with `this._restClient.documentModels.analyzeDocument` so that we don't need to cast contentType to any?

Reply:
TypeScript doesn't support the level of analysis required for that. If the `contentType` is `application/json` then `analyzeRequest` needs to be a source object. Otherwise, it needs to be a request body. The cast to any is the simplest way I could make this function work polymorphically without just duplicating the call to `analyzeDocument`.

Do you have any thoughts about how you might accomplish it?